timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java

```diff
@@ -117,6 +129,28 @@ public static TimeLockAgent create(
         return agent;
     }
 
+    private static TimeLockDialogueServiceProvider createTimeLockDialogueServiceProvider(
+            MetricsManager metricsManager, TimeLockInstallConfiguration install, UserAgent userAgent) {
+        DialogueClients.ReloadingFactory baseFactory = DialogueClients.create(
+                Refreshable.only(ServicesConfigBlock.builder().build()));
+        ServerListConfig timeLockServerListConfig = ImmutableServerListConfig.builder()
+                .addAllServers(PaxosRemotingUtils.getRemoteServerPaths(install))
+                .sslConfiguration(install.cluster().cluster().security())
+                .proxyConfiguration(install.cluster().cluster().proxyConfiguration())
+                .build();
+        return TimeLockDialogueServiceProvider.create(
+                metricsManager.getTaggedRegistry(),
+                baseFactory,
+                timeLockServerListConfig,
+                AuxiliaryRemotingParameters.builder()
+                        .userAgent(userAgent)
+                        .shouldLimitPayload(false)
+                        .shouldRetry(false) // Is subsequently overridden.
```

Comment:
We could work around this by making it Optional

Reply:
👍

Reply:
Turns out this causes a lot of unexpected headaches with the logic for detecting different remote clients, so I'm not going to block on this if that's okay?

Reply:
Fair enough, can live with this

Reply:
never mind, I coded a silly bug